src/environments.js

```diff
@@ -42,35 +120,24 @@ function parseArray(parser, result, style) {
     return new ParseNode(result.type, result, parser.mode);
 }
 
+
 /*
  * An environment definition is very similar to a function definition:
  * it is declared with a name or a list of names, a set of properties
  * and a handler containing the actual implementation.
  *
- * The properties include:
- *  - numArgs: The number of arguments after the \begin{name} function.
- *  - argTypes: (optional) Just like for a function
- *  - allowedInText: (optional) Whether or not the environment is allowed inside
- *                   text mode (default false) (not enforced yet)
- *  - numOptionalArgs: (optional) Just like for a function
- * A bare number instead of that object indicates the numArgs value.
- *
- * The handler function will receive two arguments
- *  - context: information and references provided by the parser
- *  - args: an array of arguments passed to \begin{name}
- * The context contains the following properties:
- *  - envName: the name of the environment, one of the listed names.
- *  - parser: the parser object
- *  - lexer: the lexer object
- *  - positions: the positions associated with these arguments from args.
- * The handler must return a ParseResult.
+ * - props can either be an EnvProps struct, or a number indicating the numArgs.
  */
-function defineEnvironment(names, props, handler) {
+function defineEnvironment(
+    names: string | string[],
```

Comment:
We updated `defineFucntion` to only take an array of `strings`.  We should probably do the same here.

Reply:
Done.